### PR TITLE
Update function calling docs

### PR DESCRIPTION
**Pull request by kevinjyee**
# Update Function Calling Examples

## Description of Changes

1. **Added a docstring to get_current_weather**
`litellm.utils.function_to_dict(get_current_weather)` on a function without a docstring will result in a `NoneType` exception. 

2. Made `functions` an array before passing it as a paremter into `completion`

3. Updated API key to use Anthropic API Key for the `claude` examples

### CODE REVIEW

File: docs/my-website/docs/completion/function_call.md [hunk 1]
```diff
@@ -103,11 +103,25 @@ messages = [
     {"role": "user", "content": "What is the weather like in Boston?"}
 ]
 
-def get_current_weather(location):
-  if location == "Boston, MA":
-    return "The weather is 12F"
+def get_current_weather(location: str, unit: str):
+    """Get the current weather in a given location
 
-functions = litellm.utils.function_to_dict(get_current_weather)
```
Comment (kevinjyee) [inline docs/my-website/docs/completion/function_call.md:110]: Calling `function_to_dict` on `get_current_weather` results in a NoneType exception; A docstring must be present.

File: docs/my-website/docs/completion/function_call.md [hunk 2]
```diff
@@ -103,11 +103,25 @@ messages = [
...
+    -------
+    str
+        a sentence indicating the weather
+    """
+    if location == "Boston, MA":
+        return "The weather is 12F"
+
+functions = [litellm.utils.function_to_dict(get_current_weather)]
 
 response = completion(model="gpt-3.5-turbo-0613", messages=messages, functions=functions)
```
Comment (kevinjyee) [inline docs/my-website/docs/completion/function_call.md:126]: `functions` as a parameter is of type `list`. retyped the functions variable to `list` above

File: docs/my-website/docs/completion/function_call.md [hunk 3]
```diff
@@ -103,11 +103,25 @@ messages = [
...
+
+    Returns
+    -------
+    str
+        a sentence indicating the weather
+    """
+    if location == "Boston, MA":
+        return "The weather is 12F"
+
+functions = [litellm.utils.function_to_dict(get_current_weather)]
```
Comment (kevinjyee) [inline docs/my-website/docs/completion/function_call.md:124]: Made functions to a list.

**Review by ishaan-jaff — APPROVED** [review]
lgtm!

**Comment by ishaan-jaff** [comment]
Thanks for this awesome PR @kevinjyee !

**Comment by ghost** [comment]
Amazing PR - thank you for this @kevinjyee

**Comment by ghost** [comment]
For non-openai models, our function calling can be improved, as we currently return the answer in `['choices'][0]['message']['content']` which is not how openai does it 

Screenshot 2023-10-23 at 12 09 25 PM


open to suggestions for how we can return things in a more consistent format here: @kevinjyee

**Pull request merged**